server/sonar-server/src/test/java/org/sonar/server/edition/CommitPendingEditionOnStartupTest.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * SonarQube
+ * Copyright (C) 2009-2017 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.server.edition;
+
+import org.junit.Test;
+import org.sonar.server.edition.EditionManagementState.PendingStatus;
+
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.verifyNoMoreInteractions;
+import static org.mockito.Mockito.when;
+
+public class CommitPendingEditionOnStartupTest {
+  private MutableEditionManagementState mutableEditionManagementState = mock(MutableEditionManagementState.class);
+
+  @Test
+  public void should_commit_uninstall() {
```

Comment:
behavior when LicenseCommit is installed is missing

Reply:
it's not used and should in fact never be present

Reply:
which makes my point, a test is missing

in fact, if it should never be present, then we should throw an IllegalStateException advising user to uninstall license-manager plugin (which is hard to actually code right now because of the mock of LicenseCommit always being there...)